LayoutTests/accessibility/mac/html-section-elements.html

```diff
@@ -44,8 +49,10 @@
           shouldBe("body.childAtIndex(6).childAtIndex(0).subrole", "'AXSubrole: '");
 
           // When a header and footer are inside a section, they should not be landmarks.
-          // Instead, expect only the static text inside to be exposed.
-          shouldBe("body.childAtIndex(7).role", "'AXRole: AXStaticText'");
+          shouldBe("body.childAtIndex(7).childAtIndex(0).subrole", "'AXSubrole: '");
+
+          // When a header and footer are inside a main, they should not be landmarks.
+          shouldBe("body.childAtIndex(8).childAtIndex(0).subrole", "'AXSubrole: '");
```

Comment:
This is verifying the `subrole` of static text within `section > footer`, right? If so, also probably not the thing we're trying to test.

I would expect the new testcases you added to be `childAtIndex(9)` and 10, right? If so, I don't think we are actually testing those in any way? Also it would be nice to verify their roles in addition to their subroles.

In general I don't love this test...it's hard to know what element `childAtIndex(x)` is supposed to be testing (of course this is not your fault, this problem existed before this patch). A nice improvement would be adding an `id` to the element, and then verifying the `domIdentifer()` in addition to the the role and subrole.

Reply:
You are right. The numbering is confusing. How about fixing this including the role checking in a separate patch?: 
  * Adding unique ID to the elements
  * Test their roles and subroles.